client/component/PlotComponent.tsx

```diff
@@ -285,11 +292,54 @@ class PlotComponent extends React.Component<PlotComponentProps, PlotStates> {
     } as DScatterData;
   };
 
+  append_multiline_data = (message: AppendLineDataMessage) => {
+    console.log(message);
+    const currentLineData = this.state.multilineData;
+    const newPointsData = message.al_data.map(l => this.createDLineData(l));
+    const l = Math.max(currentLineData.length, newPointsData.length);
+    let newLineData:DLineData[] = [];
+    for (let i = 0; i < l; i++) {
+      newLineData.push(this.appendDLineData(currentLineData[i], newPointsData[i]));
+    }
+    this.set_line_data(newLineData, this.state.lineAxesParams);
+  };
+
+  appendDLineData = (line: DLineData | undefined, newPoints: DLineData | null | undefined): DLineData => {
+    let con = require('ndarray-concat-rows')
+    if (newPoints === undefined || newPoints === null) {
+      return line as DLineData;
+    }
+    if (line === undefined) {
+      return this.add_indices(newPoints);
+    }
+    let x: ndarray.NdArray<TypedArray>;
+    if (!line.default_indices) {
+      if (newPoints.x.data.length === newPoints.y.data.length) {
+        x = con([line.x, newPoints.x]) as ndarray.NdArray<TypedArray>;
+      } else {
+        console.log('x and y axes must be same length ', newPoints)
+        return line as DLineData;
+      }
+    } else {
+      const len = line.y.data.length + newPoints.y.data.length;
+      let linspace = require('ndarray-linspace');
+      x = linspace(ndarray([], [len]), 0, len - 1) as ndarray.NdArray<TypedArray>;
```

Comment:
Not sure I understand this! If you are filling x values with index numbers then I expect `x = linspace(0, len-1, len)`

Reply:
It doesn't seem to work without the first argument being an ndarray and the last argument appears optional.